kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+                    }
+                }
+            }
+        }
+        for (auto& p_var_vector : mpGPVariableVectorList) {
+            const auto& r_var_database = r_gp_database.GetVariableData(*p_var_vector);
+
+            #pragma omp parallel for reduction(+:check_counter) firstprivate(result_vector)
+            for (int i = 0; i < static_cast<int>(r_elements_array.size()); ++i) {
+                auto it_elem = it_elem_begin + i;
+
+                const auto& r_entity_database = r_var_database.GetEntityData(i);
+                it_elem->CalculateOnIntegrationPoints(*p_var_vector, result_vector, r_process_info);
+                for (IndexType i_gp = 0; i_gp < result_vector.size(); ++i_gp) {
+                    for (IndexType i_comp = 0; i_comp < result_vector[i_gp].size(); ++i_comp) {
+                        const double reference = r_entity_database.GetValue(time, i_comp, i_gp);
+                        if (!CheckValues(result_vector[i_gp][i_comp], reference)) {
+                            FailMessage(it_elem->Id(), "Element", result_vector[i_gp][i_comp], reference, p_var_vector->Name(), i_comp, i_gp);
+                            check_counter += 1;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Final check
+    if (check_counter > 0) {
+        this->Set(CORRECT_RESULT, false);
+    } else {
+        this->Set(CORRECT_RESULT, true);
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FromJSONCheckResultProcess::ExecuteFinalize()
+{
+    KRATOS_TRY;
+
+    // This clears the databse of the result information
+    mDatabaseNodes.Clear();
+    mDatabaseGP.Clear();
+
+    // Set the flag to false
+    this->Set(CORRECT_RESULT, false);
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+int FromJSONCheckResultProcess::Check()
+{
+    KRATOS_TRY;
+
+    const int value_nodes = mDatabaseNodes.Check();
+    const int value_gp = mDatabaseGP.Check();
+
+    return value_nodes + value_gp;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FromJSONCheckResultProcess::InitializeDatabases()
+{
+    KRATOS_TRY;
+
+    // This clears the databse of the result information in first place
+    mDatabaseNodes.Clear();
+    mDatabaseGP.Clear();
+
+    // If we consider any flag
+    const auto& r_flag_name = mThisParameters["check_for_flag"].GetString();
+    const Flags* p_flag = (r_flag_name != "") ? KratosComponents<Flags>::Has(r_flag_name) ? &KratosComponents<Flags>::Get(r_flag_name) : nullptr : nullptr;
```

Comment:
Shouldn't we throw an error if the requested flag doesn't exist? Aside of this, why a pointer to the flag?

Reply:
Nope, flags are not mandatory in this process